README.rst

```diff
@@ -460,6 +460,154 @@ template ``context_processors`` in ``settings.py`` as shown below.
++------------------+--------------------------------------------------------------+
+| ``label``        | (``str``) Display name for the link.  (``optional``)         |
++------------------+--------------------------------------------------------------+
+| ``icon``         | (``str``) Class name of the link icon.  (``optional``)       |
++------------------+--------------------------------------------------------------+
+
+``MenuLink``
+^^^^^^^^^^^^
+
+It is used to create a link with a custom url.
+
+**Syntax:**
+
+.. code-block:: python
+
+    my_link = MenuLink(
+                config={'label': 'Link Label', 'url': 'link_url'}
+            )
+
+Following is the discription for the config
+
++------------------+--------------------------------------------------------------+
+| **Parameter**    | **Description**                                              |
++------------------+--------------------------------------------------------------+
+| ``label``        | (``str``) Display name for the link.  (``required``)         |
++------------------+--------------------------------------------------------------+
+| ``url``          | (``str``) url of the link.  (``required``)                   |
++------------------+--------------------------------------------------------------+
+| ``icon``         | (``str``) Class name of the link icon.  (``optional``)       |
++------------------+--------------------------------------------------------------+
+
+``MenuGroup``
+^^^^^^^^^^^^^
+
+It is used to create a dropdown in the menu.
+
+**Syntax:**
+
+.. code-block:: python
+
+    my_group = MenuGroup(config={'label': 'My Group Label', 'items': my_group_items, 'icon':'my-icon-class'})
+
+Following is the discription for the config
+
++------------------+--------------------------------------------------------------+
+| **Parameter**    | **Description**                                              |
++------------------+--------------------------------------------------------------+
+| ``label``        | (``str``) Display name for the link.  (``required``)         |
++------------------+--------------------------------------------------------------+
+| ``items``        | (``dict`` Items to be displayed in the dropdown.             |
+|                  | For its syntax take reference from above code example.       |
+|                  | (``required``)                                               |
++------------------+--------------------------------------------------------------+
+| ``icon``         | (``str``) Class name of the group icon.  (``optional``)      |
++------------------+--------------------------------------------------------------+
+
+**Syntax to add icon using class name:**
+
+.. code-block:: css
+
+    .icon-class-name:{
+        background: url(imageurl);
+    }
+
+**Note**: There are two ways to add links in the navigation menu i.e ``register_menu_groups`` and
```

Comment:
and we will be deprecating ``register_menu_items`` too, correct? Can we mention that information here to encourage the new functions usage?